roles/kubespray-defaults/defaults/main/main.yml

```diff
@@ -230,33 +230,57 @@ kube_pods_subnet: 10.233.64.0/18
 kube_network_node_prefix: 24
 
 # Configure Dual Stack networking (i.e. both IPv4 and IPv6)
-enable_dual_stack_networks: false
+# enable_dual_stack_networks: false # deprecated
+
+# Configure IPv4 Stack networking
+ipv4_stack: true
+# Configure IPv6 Stack networking
+ipv6_stack: "{{ enable_dual_stack_networks | default(false) }}"
 
 # Kubernetes internal network for IPv6 services, unused block of space.
-# This is only used if enable_dual_stack_networks is set to true
+# This is only used if ipv6_stack is set to true
 # This provides 4096 IPv6 IPs
 kube_service_addresses_ipv6: fd85:ee78:d8a6:8607::1000/116
 
 # Internal network. When used, it will assign IPv6 addresses from this range to individual pods.
 # This network must not already be in your network infrastructure!
-# This is only used if enable_dual_stack_networks is set to true.
+# This is only used if ipv6_stack is set to true.
 # This provides room for 256 nodes with 254 pods per node.
 kube_pods_subnet_ipv6: fd85:ee78:d8a6:8607::1:0000/112
 
 # IPv6 subnet size allocated to each for pods.
-# This is only used if enable_dual_stack_networks is set to true
+# This is only used if ipv6_stack is set to true
 # This provides room for 254 pods per node.
 kube_network_node_prefix_ipv6: 120
 
+
+kube_service_subnets: >-
```

Comment:
this should be a set_fact, like `main_ip`  ? unless we want users to tweak that variable

Reply:
this variable is composite, let the user edit individual variables, and we will control this one ourselves
Generally speaking "main_ip" is also not worth editing by user, but it's not clear how to do this

Reply:
the I think this should move out of the default and becomes a set_fact or to be put in /vars  ? 
@VannTen wdyt ?

Reply:
Yeah, agree ; variables which are not expected to be modified should live in /vars (or at task level if we only need it once, but that's not the case here right ?).

It can always be overridden by extra vars in corner cases.

I tend to avoid set_fact unless absolutely necessary, as they a task and have higher precedence.